# Ricky Doucette

## Milestone 2:

## Research question/interests

Ricky: I am interested in this dataset because I am a fan of hockey and like to see the statistical aspects of players' performances. I would like to find if there is a correlation between drops and increases in certain statistics for consecutive years for the top NHL players in points. Specific examples of research questions I would like to discover answers to are: Does an increase/decrease in time on ice per game relate to an increase in certain stats like overall points in the season? Do the top players that have a decrease in their offensive stats in a successive season make up for it with an increase in their defensive stats? How many of the top 250 players in points from the 2017-2018 season managed to make the top 350 players in points for the 2018-2019 season? How many of those players didn't make the top 350 just because they retired?

To get more specific, the research questions I would primarily like to answer are:
1. Does the average change in a top NHL player's offensive stats from one season to the next have a negative correlation with their defensive stats? That is if a player has a decrease in say points, do their stats like +/- and blocked shots tend to increase?
2. Does an increase in time on ice per game significantly increase a player's overall statistics? We could define significance as an extra point per additional minute of time on ice per game. So say a player plays 82 games both seasons. If in the first season they score 100 points with 18 minutes of TOI and the following season they score 115 points with 18.5 minutes of TOI, we could deem their increase in ice time as a significant factor in their increase in points. The calculations to determine this would be as follows:

In [3]:
import pandas as pd
nhl2017 = pd.read_csv('../data/processed/Data301ProjectData2017.csv')
nhl2018 = pd.read_csv('../data/processed/Data301ProjectData2018.csv')

Rk              Name  GP   G   A    P  PIM  +/-    TOI  PPG  GWG  PPP  \
0      1    Connor McDavid  82  41  67  108   26   20  21:32    5    7   20   
1      2     Claude Giroux  82  34  68  102   20   28  20:21    9    1   36   
2      3   Nikita Kucherov  80  39  61  100   42   15  19:48    8    7   36   
3      4     Evgeni Malkin  78  42  56   98   87   16  18:59   14    7   38   
4      5  Nathan MacKinnon  74  39  58   97   55   11  19:53   12   12   32   
..   ...               ...  ..  ..  ..  ...  ...  ...    ...  ...  ...  ...   
245  246       Devin Shore  82  11  21   32   14  -30  15:25    2    1   11   
246  247       Ryan Pulock  68  10  22   32   14   -4  18:24    5    0   11   
247  248      Noah Hanifin  79  10  22   32   21  -20  18:51    1    3    7   
248  249        Ryan Ellis  44   9  23   32    6   26  23:21    2    2    5   
249  250   Brandon Montour  80   9  23   32   42   16  20:28    5    5   12   

      G/GP   A/GP   P/GP  SHOTS     SH%  HITS  BS  
0    0.500  0.817  1.317    274  15.00%    28  46  
1    0.415  0.829  1.244    193  17.60%    30  23  
2    0.488  0.763  1.250    279  14.00%    31  15  
3    0.538  0.718  1.256    239  17.60%    48  32  
4    0.527  0.784  1.311    284  13.70%    38  22  
..     ...    ...    ...    ...     ...   ...  ..  
245  0.134  0.256  0.390    123   8.90%    71  57  
246  0.147  0.324  0.471    184   5.40%   113  86  
247  0.127  0.278  0.405    165   6.10%    81  69  
248  0.205  0.523  0.727    110   8.20%    27  84  
249  0.113  0.288  0.400    160   5.60%    96  83  

[250 rows x 19 columns]

In [1]:
s1 = (82*18)/100
s1

14.76

In [3]:
s2 = (82*18.5)/115
s2

13.191304347826087

In [4]:
s1-s2

1.568695652173913

We see here that in this example, the player was getting on average 1 point per 14.76 minutes they played in the first season, then 1 point per 13.19 minutes played in the second season. This is an increase in 1.57 points per minutes played.
If this player were to have had their TOI increased to 18.5 from 18, at the same pace of 1 point per 14.76 minutes played, they would have had about 103 points in the following season

## Milestone 3:

df.describe()